projects/plugins/jetpack/_inc/blogging-prompts.php

```diff
@@ -47,6 +47,35 @@ function jetpack_setup_blogging_prompt_response( $post_id ) {
 
 add_action( 'wp_insert_post', 'jetpack_setup_blogging_prompt_response' );
 
+/**
+ * When a published posts answers a blogging prompt, store the prompt id in the post meta.
+ *
+ * @param string  $new_status New post status.
+ * @param string  $old_status Old post status.
+ * @param WP_Post $post       Post object.
+ */
+function jetpack_mark_if_post_answers_blogging_prompt( $new_status, $old_status, $post ) {
+	// Make sure we are publishing a post, and it's not already published.
+	if ( 'post' !== $post->post_type || 'publish' !== $new_status || 'publish' === $old_status ) {
+		return;
+	}
+
+	$blocks = parse_blocks( $post->post_content );
+	foreach ( $blocks as $block ) {
+		if ( 'jetpack/blogging-prompt' === $block['blockName'] ) {
+			$prompt_id      = isset( $block['attrs']['promptId'] ) ? $block['attrs']['promptId'] : null;
```

Comment:
Should we add some sanitizing here, to be sure? I don't know if that would be overkill.

Reply:
Seems like it can't hurt! I've wrapped it with `absint`.